app/crud.py

```diff
@@ -59,9 +60,11 @@ async def create_answer_table_entry(db: Database, log_entry: schemas.LogItem):
     values = {
         k: v
         for k, v in log_entry.dict().items()
+        # :index:`question`: **Why exclude** ``act``? Some types (clickablearea, shortanswer, etc) use this field. There's probably more conditional logic needed here based on the event type.
```

Comment:
`act` is not stored in any of the answer tables, as `act` has always been a sort of catch-all field for the use info table that needs to be parsed in a unique way for each component.

ShortAnswer and Clickable should be updated to populate the answer field when they call logBookEvent even if it is just a duplicate of the act field.

Reply:
OK. We'll have to look at those cases carefully -- this also means changes to `ajax.py` until we start using the BookServer.

For the future, it would be good to better document/define what each component assigns to each query parameters in the `hsblog` endpoint -- there's currently some odd behavior you mentioned going on. (`unittest` and `timedexam` also use `act`.)

Reply:
Yep, act has always been a freeform which can be parsed based on the component.  Whereas logBookEvent just takes a javascript object that can have any fields we want populated. Yes, its a little redundant...